frontend/src/app/models/results.ts

```diff
@@ -25,24 +27,28 @@ export abstract class Results<Parameters, Result> {
      */
     loading$: Observable<boolean>;
 
-    private complete$ = new Subject<void>();
-
     constructor(
+        store: Store,
         public query: QueryModel,
-        initialParameters: Parameters,
+        protected keysInStore: string[],
     ) {
+        super(store);
+        this.connectToStore();
         this.error$ = new BehaviorSubject(undefined);
-        this.parameters$ = new BehaviorSubject(initialParameters);
 
         this.query.update.pipe(
             takeUntil(this.complete$),
             map(this.assignOnQueryUpdate.bind(this)),
         ).subscribe((params: Partial<Parameters>) =>
-            this.setParameters(params)
+            this.setParams(params)
         );
 
-        this.result$ = this.parameters$.pipe(
+        const queryUpdate$ = merge(timer(0), this.query.update);
+
+        this.result$ = combineLatest([queryUpdate$, this.state$]).pipe(
             takeUntil(this.complete$),
+            map(_.last),
```

Comment:
Why is this necessary? It's not evident from type or similar that we'll be getting in arrays here.